controllers/internal/csiscaleoperator/csiscaleoperator.go

```diff
@@ -59,14 +59,15 @@ func (c *CSIScaleOperator) GetAnnotations(daemonSetRestartedKey string, daemonSe
 		"productName":    config.ProductName,
 		"productVersion": config.DriverVersion,
 	}
-
-	if c.Annotations != nil {
-		for k, v := range c.Annotations {
-			if !labels.Has(k) {
-				labels[k] = v
+	/*
+		if c.Annotations != nil {
```

Comment:
Instead of commenting here, can we remove the annotations which are put in the CSIScaleOperator instance (c.Annotations) in the first place?

Reply:
Yes, This sounds better. Pushed a commit, give it a check.

Reply:
The annotations get's added whenever a resource is modified using `kubectl apply` command. This annotations changes values every time CR is modified from manifest file, resulting in unexpected restarts of pods. Here's a link for your reference https://kubernetes.io/docs/tasks/manage-kubernetes-objects/declarative-config/

Reply:
Looks good now, we don't have to worry about these annotations messing anywhere else now. Thanks!
Just a doubt, if we change node selector for node plugin or any other pod, are the only respective pods getting restarted when `last-applied-configuration` is removed?

Reply:
Yes.

Reply:
I think the issue of `Reconciler is restarting all the pods, even we change plugin selector label ` is not fixed with this PR. @nitishkumar4   Did you see only node plugin pod restarting after this change? I am seeing the same behavior with the latest image which we have given to test team.

```
# kc describe pod ibm-spectrum-scale-csi-operator-644b546774-5b9ml | grep "Image:"
    Image:          quay.io/madhuthorat/ibm-spectrum-scale-csi-operator-dev:goop201221

# kc apply -f tmp_driverCR.yaml
csiscaleoperator.csi.ibm.com/ibm-spectrum-scale-csi configured

# kc get pods
NAME                                               READY   STATUS    RESTARTS   AGE
ibm-spectrum-scale-csi-attacher-0                  1/1     Running   0          44s
ibm-spectrum-scale-csi-dt5zn                       3/3     Running   0          44s
ibm-spectrum-scale-csi-f44jk                       3/3     Running   0          44s
ibm-spectrum-scale-csi-operator-644b546774-5b9ml   1/1     Running   0          57s
ibm-spectrum-scale-csi-provisioner-0               1/1     Running   0          44s
ibm-spectrum-scale-csi-resizer-0                   1/1     Running   0          44s
ibm-spectrum-scale-csi-snapshotter-0               1/1     Running   0          44s

# # diff -c driverCR.yaml tmp_driverCR.yaml
*** driverCR.yaml       2021-12-28 00:25:10.905505940 -0800
--- tmp_driverCR.yaml   2021-12-28 00:36:03.948942724 -0800
***************
*** 30,36 ****
        value: "true"
    pluginNodeSelector:
      - key: "scale"
!       value: "true"
    snapshotterNodeSelector:
      - key: "scale"
        value: "true"
--- 30,36 ----
        value: "true"
    pluginNodeSelector:
      - key: "scale"
!       value: "false"
    snapshotterNodeSelector:
      - key: "scale"
        value: "true"

# kc apply -f tmp_driverCR.yaml
csiscaleoperator.csi.ibm.com/ibm-spectrum-scale-csi configured

[root@cluster2-master go]# kc get pods
NAME                                               READY   STATUS    RESTARTS     AGE
ibm-spectrum-scale-csi-attacher-0                  1/1     Running   1 (3s ago)   53s
ibm-spectrum-scale-csi-operator-644b546774-5b9ml   1/1     Running   0            66s
ibm-spectrum-scale-csi-provisioner-0               1/1     Running   1 (3s ago)   53s
ibm-spectrum-scale-csi-resizer-0                   1/1     Running   1 (3s ago)   53s
ibm-spectrum-scale-csi-snapshotter-0               1/1     Running   1 (3s ago)   53s
```